web/src/api/images/upload_image.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import * as yup from "yup";
+import { validateRequestSchema } from "src/backend/utils";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { getSdk as checkUserInAppDocumentSDK } from "@/api/images/graphql/checkUserInApp.generated";
+import { Session, getSession, withApiAuthRequired } from "@auth0/nextjs-auth0";
+import { S3Client } from "@aws-sdk/client-s3";
+import { createPresignedPost } from "@aws-sdk/s3-presigned-post";
+import { logger } from "@/lib/logger";
+
+export type ImageUploadResponse = {
+  url?: string;
+  success?: boolean;
+};
+
+const schema = yup.object({
+  app_id: yup.string().strict().required(),
+  image_type: yup
+    .string()
+    .strict()
+    .oneOf([
+      "logo_img",
+      "hero_image",
+      "showcase_img_1",
+      "showcase_img_2",
+      "showcase_img_3",
+    ])
+    .required(),
+});
+
+export type ImageUploadBody = yup.InferType<typeof schema>;
+// This endpoint takes an AppID and checks the user is on the team. If so it returns a signed URL to upload an image to S3.
+export const handleImageUpload = withApiAuthRequired(
+  async (req: NextApiRequest, res: NextApiResponse<ImageUploadResponse>) => {
+    try {
+      if (!req.method || req.method !== "POST") {
+        return errorNotAllowed(req.method, res, req);
+      }
+
+      const session = (await getSession(req, res)) as Session;
+      const auth0Team = session?.user.hasura.team_id;
+
+      const { isValid, parsedParams, handleError } =
+        await validateRequestSchema({
+          value: req.body,
+          schema,
+        });
+
+      if (!isValid || !parsedParams) {
+        return handleError(req, res);
+      }
+      const { app_id, image_type } = parsedParams;
+
+      const client = await getAPIServiceGraphqlClient();
+
+      const { team: userTeam } = await checkUserInAppDocumentSDK(
+        client
+      ).CheckUserInApp({
+        team_id: auth0Team,
+        app_id: app_id,
+      });
+      if (!userTeam[0].apps.some((app) => app.id === app_id)) {
+        return errorHasuraQuery({
+          res,
+          req,
+          detail: "User does not have access to this app.",
+          code: "no_access",
+        });
+      }
+      if (!process.env.ASSETS_S3_REGION) {
+        throw new Error("AWS Region must be set.");
+      }
+      const s3Client = new S3Client({
+        region: process.env.ASSETS_S3_REGION,
+      });
+      if (!process.env.ASSETS_S3_BUCKET_NAME) {
+        throw new Error("AWS Bucket Name must be set.");
+      }
+      const bucketName = process.env.ASSETS_S3_BUCKET_NAME;
+      const objectKey = `unverified/${app_id}/${image_type}.png`;
+      const contentType = "image/png";
```

Comment:
worth adding support for jpg. if images don't have background generally this is a more effective format